lib/internal/errors.js

```diff
@@ -116,6 +117,12 @@ const prepareStackTrace = (globalThis, error, trace) => {
   // Error: Message
   //     at function (file)
   //     at file
+  // try to avoid unnecessary getter calls
+  const code = 'code' in error ? error.code : null;
```

Comment:
Coming back to that, I don't see how it avoids a getter call:
```js
let getterCalled = false;
class MyError extends Error {
  get code() { getterCalled = true; return "CUSTOM_CODE" }
};
const error = new MyError;
console.log('code' in error); // true
const code = 'code' in error ? error.code : null;
console.log(getterCalled); // true
```
Did you mean something else? If so, can you make the comment more specific, otherwise we can just use the following:
```suggestion
  const { code } = error;
```

Reply:
@aduh95 it doesn't call the getter if the property isn't accessed:

```js
let getterCalled = false;
class MyError extends Error {
	get code() {
		getterCalled = true;
		return 'CUSTOM_CODE';
	}
}
const error = new MyError();
console.log('code' in error); // true
const code = 'code' in error ? undefined : null;
console.log(code === undefined); // true
console.log(getterCalled); // false
```

Reply:
Yes you're right - it is unnecessary. Changing.

Reply:
here's a more belligerent example using `Proxy`:

```js
/* eslint-env es6, node */
let getterCalled = false;

class MyError extends Error {
	get code() {
		getterCalled = true;
		return 'CUSTOM_CODE';
	}
}

const error = new Proxy(new MyError(), {
	get(t, p) {
		t[p];
		throw new Error();
	},
});

console.log('code' in error); // true
let code = 'code' in error ? '' : null;
console.log(code === ''); // true
console.log(getterCalled); // false

try {
	({ code } = error); // boom;
} catch {
	console.log(code); // ''
	console.log(getterCalled); // true
}
```